python/pyiceberg/expressions/literals.py

```diff
@@ -562,6 +562,13 @@ def _(self, type_var: DecimalType) -> Literal[Decimal]:
                 f"Could not convert {self.value} into a {type_var}, scales differ {type_var.scale} <> {abs(dec.as_tuple().exponent)}"
             )
 
+    @to.register(BooleanType)
+    def _(self, type_var: BooleanType) -> Literal[bool]:
+        if self.value.upper() in ['TRUE', 'FALSE']:
```

Comment:
Can we introduce a variable `value_upper` so we call `.upper()` just once?

Reply:
thanks for your suggestions, made the changes.